### PR TITLE
Add support for ovn-controller and neutron-metadata-agent split

### PR DESCRIPTION
**Pull request by luis5tb**
This depends on the edpm-ansible PR:
Depends-On: https://github.com/openstack-k8s-operators/edpm-ansible/pull/382

### CODE REVIEW

**Comment by rebtoor** [comment]
IMHO the kuttl test is failing because the ansibleee-runner image that is using is wrong.

From logs:
```
image: quay.io/openstack-k8s-operators/openstack-ansibleee-runner@sha256:64eaa1d99e0103cb4536fa9f396d41c467168cab3fe3821f8135e9a9638d7295
```

which is: https://quay.io/repository/openstack-k8s-operators/openstack-ansibleee-runner/manifest/sha256:64eaa1d99e0103cb4536fa9f396d41c467168cab3fe3821f8135e9a9638d7295

pushed on quay by this pr: https://github.com/openstack-k8s-operators/edpm-ansible/pull/375

**Comment by rebtoor** [comment]
recheck

probably the depends-on has been added after the jobs were already started because they were using the same old ansibleee runner as kuttl tests: https://review.rdoproject.org/zuul/build/0e7cb4a498394f90b648a9e7531b7d60/log/controller/ci-framework-data/logs/crc/crs/openstackansibleees.yaml

**Review by karelyatin — COMMENTED** [review]
/lgtm

**Comment by rebtoor** [comment]
recheck 

(i added the Depends-On after the job started... as for my edit in the first comment)

**Comment by rebtoor** [comment]
recheck

**Comment by karelyatin** [comment]
@rebtoor do prow kuttl jobs now supports cross repo changes with Depends-On? in past that was a limitation.

**Comment by rebtoor** [comment]
> @rebtoor do prow kuttl jobs now supports cross repo changes with Depends-On? in past that was a limitation.

I don't know, i ran into that issue as well... we should ask to CI folks

**Comment by luis5tb** [comment]
> > @rebtoor do prow kuttl jobs now supports cross repo changes with Depends-On? in past that was a limitation.
> 
> I don't know, i ran into that issue as well... we should ask to CI folks

this is the runner being used: value: quay.io/openstack-k8s-operators/openstack-ansibleee-runner@sha256:64eaa1d99e0103cb4536fa9f396d41c467168cab3fe3821f8135e9a9638d7295

**Comment by luis5tb** [comment]
recheck ERROR! Error when getting collection version metadata for community.general:5.1.1 from default (https://galaxy.ansible.com/api/) (HTTP Code: 502, Message: Bad Gateway Code: Unknown)

**Review by slagle — COMMENTED** [review]
/retest

kuttl test timed out
`
         --- FAIL: kuttl/harness/dataplane-deploy-no-nodes-test (607.73s) 
`

If it's repeatable, we may need to bump the kuttl test timeout to account for adding the extra services.

Otherwise +2, LGTM.

**Comment by booxter** [comment]
kuttl is failing because metadata job fails. but I don't understand where to find ansibleee runner pod logs to see what happened. Anyone can point me to artifacts for this? The ones linked in Prow are very limited.

**Comment by luis5tb** [comment]
@booxter @slagle there is no need to recheck the job ci/prow/dataplane-operator-build-deploy-kuttl , it is failing as depends-on is not working in prow jobs. So what happens is that the metadata job fails (as the role does not exists) and that makes that to wait until the job is claimed as Error and retried, that is why the timeout happens. The other job (zuul) where depends-on works is green

**Comment by karelyatin** [comment]
/test dataplane-operator-build-deploy-kuttl

Depends-On merged

**Comment by luis5tb** [comment]
It still used the old (8 days old) ansible-runner, instead of the one with the recent change, perhaps this requires a bump in the ansible-runner image to be used in the CVS

**Comment by karelyatin** [comment]
> It still used the old (8 days old) ansible-runner, instead of the one with the recent change, perhaps this requires a bump in the ansible-runner image to be used in the CVS

I looked into it, so what's happening is, ansibleee csv have RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT Env is set to latest runner image digest(USE_IMAGE_DIGEST is used) when it's build(and it happens after new commit merges in ansibleee repo).
So we would need a new commit merged to the repo to have that runner image reference updated to current latest. Like merging https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/pull/243 would do it.
But we would need some stable solution so in CI jobs :latest always get used. For this we may do patch ansibleee csv in install_yamls with something like:?
oc patch csv -n openstack-operators ${CSVNAME} --type='json' -p='[{"op":"replace", "path":"/spec/install/spec/deployments/0/spec/template/spec/containers/1/env/0", "value": {"name": "RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT", "value": "quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest"}}]'

**Comment by luis5tb** [comment]
/test dataplane-operator-build-deploy-kuttl

There was a build for openstack-ansibleee-operator, lets see if that is enough to get it to use the latest edpm-ansible image

**Comment by karelyatin** [comment]
> > It still used the old (8 days old) ansible-runner, instead of the one with the recent change, perhaps this requires a bump in the ansible-runner image to be used in the CVS
> 
> I looked into it, so what's happening is, ansibleee csv have RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT Env is set to latest runner image digest(USE_IMAGE_DIGEST is used) when it's build(and it happens after new commit merges in ansibleee repo). So we would need a new commit merged to the repo to have that runner image reference updated to current latest. Like merging [openstack-k8s-operators/openstack-ansibleee-operator#243](https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/pull/243) would do it. But we would need some stable solution so in CI jobs :latest always get used. For this we may do patch ansibleee csv in install_yamls with something like:? oc patch csv -n openstack-operators ${CSVNAME} --type='json' -p='[{"op":"replace", "path":"/spec/install/spec/deployments/0/spec/template/spec/containers/1/env/0", "value": {"name": "RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT", "value": "quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest"}}]'

Just to update retrigging the jobs https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/actions/runs/6394769953 refreshed the image and ansible-runner reference.

**Review by karelyatin — COMMENTED** [review]
/lgtm

**Comment by luis5tb** [comment]
retest

**Comment by karelyatin** [comment]
recheck

**Comment by gibizer** [comment]
We need this to get our tempest job back to green state https://github.com/openstack-k8s-operators/nova-operator/pull/556#issuecomment-1757805978 But hey, we have a tempest job that verifies that metadata works end to end. :)

File: config/services/dataplane_v1beta1_openstackdataplaneservice_neutron_metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneService
+metadata:
+  name: neutron-metadata
+spec:
+  label: dataplane-deployment-neutron-metadata
+  playbook: osp.edpm.neutron_metadata
+  secrets:
+  - neutron-ovn-metadata-agent-neutron-config
```
Comment (slawqo) [inline config/services/dataplane_v1beta1_openstackdataplaneservice_neutron_metadata.yaml:9]: Don't You need also nova-metadata-neutron-config here?

File: config/services/dataplane_v1beta1_openstackdataplaneservice_ovn.yaml [hunk 2]
```diff
@@ -8,5 +8,4 @@ spec:
   configMaps:
   - ovncontroller-config
   secrets:
-  - neutron-ovn-metadata-agent-neutron-config
   - nova-metadata-neutron-config
```
Comment (slawqo) [inline config/services/dataplane_v1beta1_openstackdataplaneservice_ovn.yaml:11]: this secret is needed for metadata agent, not ovn-controller

**Comment by karelyatin** [comment]
/test dataplane-operator-build-deploy-kuttl
must-gather failed

**Comment by karelyatin** [comment]
recheck NODE_FAILURE

**Pull request merged**